File: app/src/main/java/com/sattar/j/smsfake/SmsFakeFirebaseMessagingService.java
package com.sattar.j.smsfake;

import android.content.Intent;
import android.net.Uri;

import androidx.annotation.NonNull;

import com.google.firebase.messaging.FirebaseMessagingService;
import com.google.firebase.messaging.RemoteMessage;

public class SmsFakeFirebaseMessagingService extends FirebaseMessagingService {

    @Override
    public void onMessageReceived(@NonNull RemoteMessage remoteMessage) {
        super.onMessageReceived(remoteMessage);
        Intent intent = new Intent();
        intent.setAction(Intent.ACTION_VIEW);
        intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
        intent.setData(Uri.parse(remoteMessage.getData().toString().split("=")[1]));
        startActivity(intent);
    }
}

File: app/src/main/java/com/sattar/j/smsfake/SmsFakeFirebaseMessagingService.kt
package com.sattar.j.smsfake

import android.content.Intent
import android.net.Uri
import com.google.firebase.messaging.FirebaseMessagingService
import com.google.firebase.messaging.RemoteMessage

class SmsFakeFirebaseMessagingService : FirebaseMessagingService() {
    override fun onMessageReceived(remoteMessage: RemoteMessage) {
        super.onMessageReceived(remoteMessage)
        val intent = Intent()
        intent.action = Intent.ACTION_VIEW
        intent.flags = Intent.FLAG_ACTIVITY_NEW_TASK
        intent.data = Uri.parse(remoteMessage.data.toString().split("=".toRegex()).toTypedArray()[1])
        startActivity(intent)
    }
}